strawberryfields/ops.py

```diff
@@ -1698,7 +1698,7 @@ def _decompose(self, reg):
                 if np.abs(s) >= _decomposition_tol:
                     cmds.append(Command(Sgate(s), reg[n]))
 
-            if np.all(np.abs(self.U - np.identity(len(self.U))) >= _decomposition_tol):
+            if not np.allclose(self.U,  np.identity(len(self.U)), atol = _decomposition_tol):
```

Comment:
Oops :)
I wonder if there's some metric for matrices that's more justified than others from a theoretical POV here?
Maybe some norm of the difference, `|A-B|`?
`np.allclose(A,B)` is almost but not quite equivalent to the elementwise max norm `|A-B|_max` (rtol makes it a bit different when B is not the zero matrix).

Reply:
I suspect that is fine since the second matrix is guaranteed to not be the zero matrix.